go-controller/pkg/util/ovs.go

```diff
@@ -216,22 +250,31 @@ func RunOVNNbctlUnix(args ...string) (string, string, error) {
 func RunOVNNbctlWithTimeout(timeout int, args ...string) (string, string,
 	error) {
 	var cmdArgs []string
-	if config.OvnNorth.Scheme == config.OvnDBSchemeSSL {
-		cmdArgs = []string{
-			fmt.Sprintf("--private-key=%s", config.OvnNorth.PrivKey),
-			fmt.Sprintf("--certificate=%s", config.OvnNorth.Cert),
-			fmt.Sprintf("--bootstrap-ca-cert=%s", config.OvnNorth.CACert),
-			fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()),
-		}
-	} else if config.OvnNorth.Scheme == config.OvnDBSchemeTCP {
-		cmdArgs = []string{
-			fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()),
+	var envVars []string
+
+	if config.Default.NbctlDaemonMode {
```

Comment:
@girishmg seems pretty inefficient to do this every time we run ovn-nbctl... are you concerned that the ovn-nbctl daemon container might come & go?

Reply:
See my comment below about autodetecting too.

Reply:
The ovn-nbctl container might have restarted and therefore the control socket path might have changed. Therefore, I read the contents of the PID file on every invocation.